cuda_core/tests/test_event.py

```diff
@@ -6,34 +6,46 @@
 # this software and related documentation outside the terms of the EULA
 # is strictly prohibited.
 
-from cuda import cuda
-from cuda.core.experimental._event import  EventOptions, Event
-from cuda.core.experimental._utils import handle_return
+from cuda.core.experimental._event import  EventOptions
 from cuda.core.experimental._device import Device
-import pytest
 
-def test_is_timing_disabled():
+def test_is_timing_disabled(init_cuda):
     options = EventOptions(enable_timing=False)
-    event = Event._init(options)
+    stream = Device().create_stream()
+    event = stream.record(options=options)
     assert event.is_timing_disabled == True
+    
+    options = EventOptions(enable_timing=True)
+    stream = Device().create_stream()
+    event = stream.record(options=options)
+    assert event.is_timing_disabled == False
 
-def test_is_sync_busy_waited():
-    options = EventOptions(busy_waited_sync=True)
-    event = Event._init(options)
+def test_is_sync_busy_waited(init_cuda):
+    options = EventOptions(enable_timing=False, busy_waited_sync=True)
+    stream = Device().create_stream()
+    event = stream.record(options=options)
     assert event.is_sync_busy_waited == True
 
-def test_sync():
-    options = EventOptions()
-    event = Event._init(options)
+    options = EventOptions(enable_timing=False)
+    stream = Device().create_stream()
+    event = stream.record(options=options)
+    assert event.is_sync_busy_waited == False
+
+def test_sync(init_cuda):
+    options = EventOptions(enable_timing=False)
+    stream = Device().create_stream()
+    event = stream.record(options=options)
     event.sync()
     assert event.is_done == True
 
-def test_is_done():
-    options = EventOptions()
-    event = Event._init(options)
+def test_is_done(init_cuda):
+    options = EventOptions(enable_timing=False)
+    stream = Device().create_stream()
+    event = stream.record(options=options)
     assert event.is_done == True
 
-def test_handle():
-    options = EventOptions()
-    event = Event._init(options)
+def test_handle(init_cuda):
+    options = EventOptions(enable_timing=False)
+    stream = Device().create_stream()
+    event = stream.record(options=options)
     assert isinstance(event.handle, int)

```

Comment:
nit: I think we should just move the assert to one of the tests. The test body overlaps with other tests.